app/components/common/emailChipsInput/index.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-useless-escape */
+import React, { FC, useState } from 'react';
+
+interface EmailChipsInputProps {
+  label: string;
+  placeholder: string;
+  attendees: string[];
+  setAttendees: (att: string[]) => void;
+}
+
+const EmailChipsInput: FC<EmailChipsInputProps> = ({
+  label,
+  placeholder,
+  attendees,
+  setAttendees,
+}) => {
+  const [value, setValue] = useState('');
+  const [error, setError] = useState<string | null>(null);
+
+  const isInList = (email: string) => attendees.includes(email);
+
+  const isEmail = (email: string) => /[\w\d\.-]+@[\w\d\.-]+\.[\w\d\.-]+/.test(email);
+
+  const isValid = (email: string) => {
+    let err = null;
+
+    if (isInList(email)) {
+      err = `${email} has already been added.`;
+    }
+
+    if (!isEmail(email)) {
+      err = `${email} is not a valid email address.`;
+    }
+
+    if (err) {
+      setError(err);
+      return false;
+    }
+
+    return true;
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
```

Comment:
I can see more logic present in the UI component. What we can do is extract all the logic part as a hook and keep this a pure component. Hook name can be useEmailChipsInput

Reply:
@DeRaowl thanks for the suggestion, but, the extraction part will not be beneficial as the logic is closely knit for this component and wont be reused in other components.